plugins/modules/vault_pki_generate_certificate.py

```diff
@@ -0,0 +1,282 @@
+      returned: success
+      type: complex
+      contains:
+        certificate:
+          description: Generated certificate.
+          returned: success
+          type: str
+          sample: "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----"
+        issuing_ca:
+          description: CA certificate.
+          returned: success
+          type: str
+          sample: "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----"
+        ca_chain:
+          description: Linked list of CA certificates.
+          returned: success
+          type: list
+          elements: str
+          sample: ["-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----"]
+        private_key:
+          description: Private key used to generate certificate.
+          returned: success
+          type: str
+          sample: "-----BEGIN RSA PRIVATE KEY-----...-----END RSA PRIVATE KEY-----"
+        private_key_type:
+          description: Private key algorithm.
+          returned: success
+          type: str
+          sample: rsa
+        serial_number:
+          description: Certificate's serial number.
+          returned: success
+          type: str
+          sample: 39:dd:2e:90:b7:23:1f:8d:d3:7d:31:c5:1b:da:84:d0:5b:65:31:58
+    warning:
+      description: Warnings returned by Vault during generation.
+      returned: success
+      type: str
+"""
+
+import traceback
+
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import missing_required_lib
+
+from ansible_collections.community.hashi_vault.plugins.module_utils._hashi_vault_module import HashiVaultModule
+from ansible_collections.community.hashi_vault.plugins.module_utils._hashi_vault_common import HashiVaultValueError
+
+HAS_HVAC = False
+try:
+    import hvac
+    from hvac.api.secrets_engines.pki import DEFAULT_MOUNT_POINT
+    HAS_HVAC = True
+except ImportError:
+    HVAC_IMPORT_ERROR = traceback.format_exc()
+    HAS_HVAC = False
+
+
+def run_module():
+    argspec = HashiVaultModule.generate_argspec(
+        role_name=dict(type='str', required=True),
+        common_name=dict(type='str', required=True),
+        alt_names=dict(type='list', elements='str', required=False, default=[]),
+        ip_sans=dict(type='list', elements='str', required=False, default=[]),
+        uri_sans=dict(type='list', elements='str', required=False, default=[]),
+        other_sans=dict(type='list', elements='str', required=False, default=[]),
+        ttl=dict(type='str', required=False, default=None),
+        format=dict(type='str', required=False, choices=['pem', 'der', 'pem_bundle'], default='pem'),
+        private_key_format=dict(type='str', required=False, choices=['der', 'pkcs8'], default='der'),
+        exclude_cn_from_sans=dict(type='bool', required=False, default=False),
+        engine_mount_point=dict(type='str', required=False)
+    )
+
+    module = HashiVaultModule(
+        argument_spec=argspec,
+        supports_check_mode=True
```

Comment:
I have added the requisite lines for it to support check mode (see 263).

If you mean that it is not idempotent in the sense that it may perform a login, that is true and something to consider across the whole collection. I am starting to add such information to our docsite, you may review #211 if you'd like to see some of that!

---

For this module in particular, even though the status will always be `changed`, I believe that supporting check mode is more valuable than not. If we decline check mode support, Ansible will skip this module in check mode, and so it will show a skipped status.

That's good for safety, one of the two big reasons to use check mode, it's not so good for the other, which is to _report_ on what will change and what will not. By supporting check mode, we provide a `changed` status which is (at least slightly) more useful, in my opinion.